cmd/releasego/update-azure-linux.go

```diff
@@ -96,19 +107,79 @@ func updateAzureLinux(p subcmd.ParseFunc) error {
 		return err
 	}
 
-	ar := txtar.Archive{
-		Comment: []byte("Bump version to " + assets.GoVersion().Full()),
-		Files: []txtar.File{
-			{Name: cgManifestFilepath, Data: cgManifestBytes},
-			{Name: golangSignaturesFilepath, Data: golangSignaturesFileBytes},
-			{Name: golangSpecFilepath, Data: []byte(golangSpecFileContent)},
-		},
+	ref, _, err := client.Git.GetRef(ctx, owner, repo, baseBranch)
+	if err != nil {
+		return fmt.Errorf("Failed to get ref: %v", err)
+	}
+
+	newRef := &github.Reference{
+		Ref:    github.String(updateBranch),
+		Object: &github.GitObject{SHA: ref.Object.SHA},
+	}
+
+	if _, _, err = client.Git.CreateRef(ctx, owner, repo, newRef); err != nil {
+		return fmt.Errorf("Failed to create ref: %v", err)
+	}
+
+	updatedFiles := map[string][]byte{
+		golangSpecFilepath:       []byte(golangSpecFileContent),
+		golangSignaturesFilepath: golangSignaturesFileBytes,
+		cgManifestFilepath:       cgManifestBytes,
+	}
+
+	for filePath, newContent := range updatedFiles {
+		// Get the file to update
+		file, _, _, err := client.Repositories.GetContents(ctx, owner, repo, filePath, &github.RepositoryContentGetOptions{Ref: updateBranch})
+		if err != nil {
+			return fmt.Errorf("Failed to get file %q: %v", filePath, err)
+		}
+
+		// Update the file
+		_, _, err = client.Repositories.UpdateFile(ctx, owner, repo, filePath, &github.RepositoryContentFileOptions{
```

Comment:
This would make a separate commit per file, right? Could we use https://pkg.go.dev/github.com/google/go-github/github#GitService.CreateTree -> CreateCommit -> UpdateRef instead? (I've done it before, but from C# code.)

Reply:
After discussion with azure linux team member we decided to go with squash and merge, this removes a  need for making everything in single commit.